sql/core/src/main/scala/org/apache/spark/sql/execution/joins/HashJoin.scala

```diff
@@ -214,6 +215,7 @@ trait HashJoin {
     val resultProj = createResultProjection
     joinedIter.map { r =>
       numOutputRows += 1
+      avgHashProbe.set(hashed.getAverageProbesPerLookup())
```

Comment:
Because in HashAggregate, we process all rows in aggregation, then we can update the probe. But in join, it processes the rows as an iterator, we have no a point to update the probe in the end of join like aggregation case.